tests/kani/Intrinsics/bitreverse.rs

```diff
@@ -38,10 +30,11 @@ fn main() {
     test_bitreverse_intrinsic!(u16, check_reverse_u16, get_bit_at_u16);
     test_bitreverse_intrinsic!(u32, check_reverse_u32, get_bit_at_u32);
     test_bitreverse_intrinsic!(u64, check_reverse_u64, get_bit_at_u64);
+    test_bitreverse_intrinsic!(u128, check_reverse_u128, get_bit_at_u128);
     test_bitreverse_intrinsic!(usize, check_reverse_usize, get_bit_at_usize);
-    test_bitreverse_intrinsic!(i8, check_reverse_i8, get_bit_at_i8);
-    test_bitreverse_intrinsic!(i16, check_reverse_i16, get_bit_at_i16);
-    test_bitreverse_intrinsic!(i32, check_reverse_i32, get_bit_at_i32);
-    test_bitreverse_intrinsic!(i64, check_reverse_i64, get_bit_at_i64);
-    test_bitreverse_intrinsic!(isize, check_reverse_isize, get_bit_at_isize);
+    // `reverse_bits` is also defined for signed integer types by casting the
+    // number into the corresponding unsigned type and then casting the result
+    // into the original signed type. This causes overflows unless we restrict
+    // their values considerably, making the signed versions not very
+    // interesting to test here.
```

Comment:
If bit reverse works on signed types in rust... shouldn't we test that we translate it correctly still?

Reply:
Well, the translation is the same. Just to be clear, this is how it is defined: https://github.com/rust-lang/rust/blob/master/library/core/src/num/int_macros.rs#L277-L279

Would you be happy if we add a test like this for each integer type?
```rust
assert!((0 as i8).reverse_bits() == 0);
```